### ISSUE TITLE
Extract archive with a folder at his root not working

### ISSUE THREAD

**Issue by olivierbossel** [wapmorgan/UnifiedArchive#11]
**Describe the bug**
When I try to extract an archive created with a folder as source, I got this error back : 
```
Warning: ZipArchive::extractTo(/Users/olivierbossel/data/web/dwarvesforge/thorin/data/extracted/): failed to open stream: Is a directory in /Users/olivierbossel/data/web/dwarvesforge/thorin/vendor/wapmorgan/unified-archive/src/UnifiedArchive.php on line 689
```

**Configration (please complete the following information):**
 - OS: OSX
 - Version: 10.13.6

**To Reproduce**
1. Zip a folder with an image inside like so:
```php
UnifiedArchive::archiveFiles('my-folder', 'my-archive.zip');
```
2. Unzip the created archive like so:
```php
$archiveInstance = UnifiedArchive::open('my-archive.zip');
$archiveInstance->extractFiles('folder-to-extract-to');
```

**Expected behavior**
I expect that my archive will be extracted and that I get back a folder structure like so:
| folder-to-extract-to
|--- my-folder
|------ my-image.jpg

Thanks in advance! And thanks for your nice job on this library!
Cheers!

**Comment by wapmorgan** [comment]
Hello. Which version of UA used?

**Comment by olivierbossel** [comment]
Hi.
Thanks for your feedback.
I use the 0.1.1

**Comment by wapmorgan** [comment]
It seems to be [a directory "/" inside the archive](https://stackoverflow.com/questions/32157272/ziparchive-failed-to-open-stream-is-a-directory).

If it was created with UA, I'll discover this issue.

**Comment by wapmorgan** [comment]
Confirm that this made by UA **0.1.1.** Unfortunately, in this version also present bug with extraction. 

So I'd propose switch to **0.1.x-dev** branch, where:
1. When you create archive, "/" entry is not added, so no problems at all.
2. To unpack an archive with "/" entry, use this code:
```php
$ar = wapmorgan\UnifiedArchive\UnifiedArchive::open('my-archive6.zip');
$ar->extractFiles('output-folder', array_values(array_filter($ar->getFileNames(), function ($val) { return $val !== '/'; })));
```

**Comment by olivierbossel** [comment]
Hi man,
Thanks for the reply. I will wait for the fix in the next release.
Cheers!

**Comment by wapmorgan** [comment]
Released as 0.1.2.
